src/coreclr/jit/emitxarch.cpp

```diff
@@ -9578,6 +9620,11 @@ const char* emitter::emitRegName(regNumber reg, emitAttr attr, bool varName)
 #ifdef TARGET_AMD64
     char suffix = '\0';
 
+    if (isMaskReg(reg))
+    {
+        return emitKregName(reg);
```

Comment:
I don't think you need `emitKregName` at all.
```suggestion
        return rn;
```

Any direct callers can just use `getRegName()` instead.